### DISCUSSION TITLE
Markor content and git

### DISCUSSION THREAD

**Discussion by aisbergde** [gsantner/markor#1595] [Q&A] [answered]
I use my notes system on the PC and on Android. In the past I used Onedrive to sync, but then I moved to use a private git repository on GitLab. It would also be possible to use GitHub, but for many reasons I prefer GitLab over GitHub, when possible.

I found only one Android git client, which is still developed: mgit. It works quite good, but staging and committing changes takes very long and there is no information about the progress. But it works. And this way I can keep my changes under git version control.

I also found 2 editors with build in GitLab support: They can connect directly to GitLab repositories.

- Code Editor does a commit on each save, and I really like this.
- Spck Editor has more common git features like push, pull, commit. It is also fine

I want to discuss:

- Does someone has other experience, how to use Markor with git?
- Are there more easy git clients with more performant staging and commiting?
- Would it be a good idea to have a simple built in git client directly in Markor like one of the other editors?

I also tried gitjournal. But it is too focused on Markdown and I switched from Markdown to AsciiDoc.

**Comment by aisbergde** [comment]
Or maybe someone knows a file manager with direct GitLab integration?

**Reply by gsantner** [reply]
A combination of 

* `<whatever software>` to directly edit files in i.e. a gitlab repo's Wiki, via the API 
* AND Markor + some git application for the rest

is maybe an opion.

**Comment by gsantner** [comment, marked as answer]
I have git somewhere at the end, not directly as you would think.

* On phone edit with Markor
* Sync files with syncthing, including i.e. my own kind of Wikipedia with articles written in markdown
* On another single device where syncthing is running, the files get automatically synced
* On that device, I also automatically git commit & git push via cron job.
* TL;DR Nice & hasslefree  synced AND I have automatically a full git history.

**Reply by aisbergde** [reply]
this is also interesting, to use the "server" as a Git client, as you discribe. For example a NAS could be used. This also should avoid sync conflicts.
My solution to use only Onedrive sometimes went into a conflict, when the syncing didn't work properly and I ended in 2 files. That's why the idea to use git.
I should think about this approach.

**Reply by gsantner** [reply]
I can't tell you what todo and whats perfect. But for the many things I tried, Syncthing for most files + git on top for version-able files turned out to be the best option so far.

I had lots of sync conflict issues with let's say NextCloud as well - guess thats quite similar to what you run to at Onedrive. At Syncthing I rarely have them, and even if so - it still syncs without interaction, and the sync-conflict files are synced to all devices. So I can eliminate sync conflicts later (i.e. by diff) on any device, when I have time for it - and I can this way still add new todos on any device without worrying about loosing anything.

**Comment by X7md** [comment]
So what about Termux + Termux:Widget to upload a file to your remote git server?

**Reply by aisbergde** [reply]
I never used Termux. But Termux:Widget could be interesting, if Termux contains a Git client and I just need to prepare some scripts containing some git commands. I will investigate.

As a typical Windows user I use GUI whenever possible, but this is limitating ways to solve issues.
